lms/services/course.py

```diff
@@ -78,7 +78,7 @@ def search(  # noqa: PLR0913, PLR0917
         name: str | None = None,
         limit: int = 100,
         organization_ids: list[int] | None = None,
-        user: User | None = None,
+        h_userid: str | None = None,
```

Comment:
I think the previous design, passing "User" and using "User.h_userid" could be confusing.

If we are filtering by h_userid, just pass that.